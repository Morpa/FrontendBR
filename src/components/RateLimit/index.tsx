import * as S from './styles'

export type RateLimitProps = {
  title: string
  description: string
}

const RateLimit = ({ title, description }: RateLimitProps) => (
  <S.Wrapper>
    <S.Image
      src="/img/limit.png"
      alt="Um computador com um triÃ¢ngulo vermelho sobre ele"
      role="img"
    />

    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
  </S.Wrapper>
)

export default RateLimit
