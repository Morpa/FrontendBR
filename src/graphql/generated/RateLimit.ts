/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: RateLimit
// ====================================================

export interface RateLimit_rateLimit {
  __typename: "RateLimit";
  limit: number;
  remaining: number;
}

export interface RateLimit {
  rateLimit: RateLimit_rateLimit | null;
}
