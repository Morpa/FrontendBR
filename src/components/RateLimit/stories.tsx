import { Story, Meta } from '@storybook/react/types-6-0'
import RateLimit, { RateLimitProps } from '.'

export default {
  title: 'RateLimit',
  component: RateLimit
} as Meta

export const Default: Story<RateLimitProps> = (args) => <RateLimit {...args} />

Default.args = {
  title: 'Opss...',
  description:
    'Parece que você atingiu o limite de requisições. Aguarde alguns minutos...'
}
