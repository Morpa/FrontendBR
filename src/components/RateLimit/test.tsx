import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import RateLimit from '.'

const props = {
  title: 'A simple title',
  description: 'A simple description'
}

describe('<RateLimit />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<RateLimit {...props} />)

    expect(
      screen.getByRole('img', {
        name: /Um computador com um tri√¢ngulo vermelho sobre ele/i
      })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /a simple title/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/a simple description/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
