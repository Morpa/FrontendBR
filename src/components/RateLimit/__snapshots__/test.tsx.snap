// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RateLimit /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c1 {
  max-width: 100%;
  margin-top: 1.6rem;
}

.c2 {
  color: #F231A5;
  font-size: 2.8rem;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

<main
  class="c0"
>
  <img
    alt="Um computador com um triÃ¢ngulo vermelho sobre ele"
    class="c1"
    role="img"
    src="/img/limit.png"
  />
  <h2
    class="c2"
  >
    A simple title
  </h2>
  <p
    class="c3"
  >
    A simple description
  </p>
</main>
`;
